### PR TITLE
Responding to MX queries

### PR DESCRIPTION
**Pull request by andrenatal**

### CODE REVIEW

File: server/src/pdns.rs [hunk 1]
```diff
@@ -215,6 +215,22 @@ fn process_request(req: PdnsRequest, config: &Config) -> Result<PdnsResponse, St
 
         debug!("final qname={}", qname);
 
+        if qtype == "ANY" {
+            // Add a "MX" record.
+            let ns_record = PdnsLookupResponse {
+                qtype: "MX".to_owned(),
+                qname: original_qname.to_owned(),
+                content: "10 inbound-smtp.us-west-2.amazonaws.com".to_owned(),
```
Comment (fabricedesre) [inline server/src/pdns.rs]: that should not be hardcoded. Make it an optional config field instead.

**Comment by andrenatal** [comment]
@fabricedesre thanks, that was a provisory patch. This is the final version. Could you take another look, please?

File: server/src/email_routes.rs [hunk 2]
```diff
@@ -139,10 +139,9 @@ pub fn setemail(req: &mut Request, config: &Config) -> IronResult<Response> {
                         Some(_) => "https",
                         None => "http",
                     };
-                    let full_link = format!("{}://{}:{}/confirmemail?s={}",
+                    let full_link = format!("{}://api.{}/verifyemail?s={}",
                                             scheme,
                                             config.options.general.domain,
-                                            config.options.general.port,
```
Comment (fabricedesre) [inline server/src/email_routes.rs]: That change looks unrelated to this PR, but in any case: what happens if you don't run on port 80 or 443?

File: server/src/config.rs [hunk 3]
```diff
@@ -21,6 +21,8 @@ pub struct PdnsOptions {
     pub soa_content: String,
     pub socket_path: Option<String>,
     pub dns_ttl: u32,
+    pub mx_record: String,
```
Comment (fabricedesre) [inline server/src/config.rs]: That should be an `Option<String>` no? why mandate pdns to be the authoritative source for the MX field?

File: config/config.toml [hunk 4]
```diff
@@ -24,7 +26,7 @@ success_page = """<!DOCTYPE html>
 <html>
   <head><title>Email Confirmation Successful!</title></head>
   <body>
-    <h1>Thank you for verifying your email, {email}.</h1>
+    <h1>Thank you for verifying your email.</h1>
   </body>
```
Comment (fabricedesre) [inline config/config.toml]: is that change related to this PR?

File: server/src/pdns.rs [hunk 5]
```diff
@@ -290,14 +306,12 @@ fn process_request(req: PdnsRequest, config: &Config) -> Result<PdnsResponse, St
                         .result
                         .push(PdnsResponseParams::Lookup(ns_record));
                 }
-
-                return Ok(pdns_response);
             }
             Err(_) => {
-                // No such domain, return a `false` result to PowerDNS.
-                return Err("No record for this name.".to_owned());
+                info!("No record for this name {}", qname);
```
Comment (fabricedesre) [inline server/src/pdns.rs]: I am not convinced that this correct to not send an error here anymore. Can you explain the logic change here?

File: server/src/email_routes.rs (hunk 2 above)
Comment (andrenatal) [reply server/src/email_routes.rs]: We changed the architecture and now we have an nginx proxy distributing the requests to api.* and for pagekite to allow all of them to come through port 80 or 443

File: server/src/pdns.rs (hunk 5 above)
Comment (andrenatal) [reply server/src/pdns.rs]: We need an MX record to always be responded (since we need to receive emails in the tld) even if the qname does not exist in the database. 

The situation I faced when testing while having this error being returned to pdns, was that when an MX query was made, it never responded the DNS query properly, because since the qname wasn't found in the db, we had `Err("No record for this name.".to_owned());` always being sent in the end, instead of `Ok(pdns_response);` consequently making the SMTP unable to reach the MX server and ultimately returning the email.

If I had at least the correct query type being attributed to`qtype`  (it always only have `SOA` and `ANY`) I could overcome this and return immediately if the request is an `MX` but I'm not sure if something need to be changed on powerdns or if that's the normal behavior. 

This approach was tested on staging and didn't affect neither powerdns, pagekite and the API.

Comment (andrenatal) [reply server/src/pdns.rs]: This approach was tested on staging and didn't affect neither powerdns, pagekite and the API.

Comment (andrenatal) [reply server/src/pdns.rs]: Another situation we faced that illustrates this use case was we needed to add a TXT challenge record to the TLD to allow amazon to handle the domain's SMTP. With the  `Err("No record for this name.".to_owned());` being returned when the `qname` isn't found we never had the challenge validated because nothing except an error was returned all the time.
